test/extensions/filters/common/expr/context_test.cc

```diff
@@ -391,6 +391,17 @@ TEST(Context, ResponseAttributes) {
     auto value = response_no_status[CelValue::CreateStringView(GrpcStatus)];
     EXPECT_FALSE(value.has_value());
   }
+  {
+    // The response header doesn't contain :status but the streamInfo has one.
```

Comment:
I'm not sure if this can ever happen in practice, a response without a :status would fail somewhere (not quite sure where). Normally when code reads :status from the response headers it will just assume it's there for this reason

Reply:
yeah I'd rather fix any disparity in these two values than work around it in one location in the code base.

Reply:
@alyssawilk WDYT the semantics of StreamInfo::statusCode should be? This PR originally set it during `ActiveStreamDecoderFilter::encodeHeaders` but that doesn't account for possible mutations made to :status on the encoding path. I'd suggested just having `StreamInfo::statusCode` read from the response headers instead of being its own field, but it depends on whether we want `StreamInfo::statusCode` to contain the original code or the most recently updated one

Reply:
I would think we'd want the most recent one.  I'd be inclined to remove it entirely from stream info and have a helper function if we can, else refactor as you suggest.